File: examples/python/gui_demo.py
import visual_novel_engine as vn

script_json = """
{
  "events": [
    {"type": "dialogue", "speaker": "System", "text": "Bienvenido a la demo gráfica."}
  ],
  "labels": {"start": 0}
}
"""

config = vn.VnConfig(width=1024.0, height=768.0, fullscreen=False)

print("Iniciando novela visual...")
try:
    vn.run_visual_novel(script_json, config)
    print("Ejecución finalizada.")
except Exception as exc:
    print(f"Error: {exc}")

File: examples/gui_demo.rs
use visual_novel_gui::{run_app, VnConfig};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let script_json = r#"
    {
      "events": [
        {"type": "dialogue", "speaker": "System", "text": "Bienvenido a la demo gráfica."}
      ],
      "labels": {"start": 0}
    }
    "#;

    let config = VnConfig {
        title: "Demo Visual Novel".to_string(),
        width: Some(1280.0),
        height: Some(720.0),
        ..Default::default()
    };

    run_app(script_json.to_string(), Some(config))?;
    Ok(())
}
